R/nba_ranking.R

```diff
@@ -1,19 +1,97 @@
 #' Generates a ranking and a visualization based on a column of a dataset
 #'
 #' @description
-#' This function generates a ranking and a visualization for a variable, according to a different variable the user sets.
+#' This function generates creates a ranking of a variable `column` summarizing by a variable `by` using a function `FUN`. The result of this function is
+#' a visualization with that information.
 #'
 #' @param data The dataframe to make the ranking from
 #' @param column The column from the dataset to rank
 #' @param by The column from the dataset to rank by
 #' @param top The number of elements in the ranking
+#' @param descending Boolean variable for the order of the ranking. TRUE if descending, FALSE otherwise.
+#' @param FUN function to apply to the values
 #'
 #' @return ggplot visualization with the ranking
 #'
 #' @export
 #'
+#' @import dplyr
+#' @import forcats
+#' @import ggplot2
+#' @import scales
+#' @importFrom magrittr %>%
+#' @importFrom rlang :=
+#'
 #' @examples
-#' nba_ranking(nba_data, 'Player', 'TD3', 10)
-nba_ranking <- function(data, column, by, top){
-  return (1)
+#' nba_ranking(data.frame(ranked = c(1, 2, 3), by = c(3, 2, 1)), ranked, by, 2, TRUE, mean)
+nba_ranking <- function(data, column, by, top, descending = TRUE, FUN){
+
+  # Checks
+
+  # Checking if the top variable has the desired type
+  if(!typeof(top) == 'double'){
+    stop("Argument top should be a numerical value")
+  }
+
+  # Checking if the data is a dataframe
+  if(!typeof(data) == 'list'){
+    stop("Argument data should be a dataframe or a tibble")
+  }
+
```

Comment:
Add an exception handling for column variable to only take in character columns. Does not make sense to take in a numeric value here.

Reply:
if(!typeof(column) == 'character'){
    stop("Argument column should be a categorical value")
  }